src/SFA.DAS.ApprenticeCommitments/Data/FuzzyMatching/FuzzyMatcher.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using F23.StringSimilarity;
+using NinjaNye.SearchExtensions.Soundex;
+
+namespace SFA.DAS.ApprenticeCommitments.Data.FuzzyMatching
+{
+    public class FuzzyMatcher
+    {
+        public int _similarityThreshold { get; set; } = 0;
+
+        public FuzzyMatcher(int similarityThreshold)
+            => _similarityThreshold = similarityThreshold;
+
+        public bool IsSimilar(string string1, string string2)
+        {
+            if (string1 == string2)
+            {
+                return true;
+            }
+
+            string1 = string1.Trim();
+            string2 = string2.Trim();
+
+            if (string1.Contains(" ") || string2.Contains(" ") || string1.Contains("-") || string2.Contains("-"))
+            {
+                foreach (var separator in new List<char>() { ' ', '-' })
+                {
+                    // lets first remove the char and do a match, so "harry" should match "ha r r y"   
+                    if (GetSimilarity(string1.Replace(separator.ToString(), ""), string2.Replace(separator.ToString(), "")) >= _similarityThreshold)
+                    {
+                        return true;
+                    }
+                }
+
+                if (CheckSeperators(string1, string2))
+                {
+                    return true;
+                }
+            }
+
+            if (GetSimilarity(string1, string2) >= _similarityThreshold)
+            {
+                return true;
+            }
```

Comment:
I think you can replace pretty much all this manual looping (and the methods below) with a couple of `Any` calls:

```suggestion
            var s1 = string1.Split(' ', '-');
            var s2 = string2.Split(' ', '-');

            if (s1.Any(s => s2.Any(r => GetSimilarity(s, r) >= _similarityThreshold)))
                return true;

```